zebedee-cms/src/main/java/com/github/onsdigital/zebedee/keyring/LegacyKeyringImpl.java

```diff
@@ -313,29 +320,57 @@ public void unlock(User user, String password) throws KeyringException {
     }
 
     @Override
-    public void populate(User src, User target, Set<String> collectionIDs) throws KeyringException {
+    public void assignTo(User src, User target, List<CollectionDescription> assignments) throws KeyringException {
+        if (assignments == null || assignments.isEmpty()) {
+            return;
+        }
+
         validateUser(src);
+        validateUser(target);
 
-        com.github.onsdigital.zebedee.json.Keyring srcKeyring = getCachedUserKeyring(src);
-        if (srcKeyring == null) {
+        com.github.onsdigital.zebedee.json.Keyring srcCache = getCachedUserKeyring(src);
+        if (srcCache == null) {
             throw new KeyringException(CACHE_KEYRING_NULL_ERR);
         }
 
-        if (!srcKeyring.isUnlocked()) {
+        if (!srcCache.isUnlocked()) {
             throw new KeyringException(KEYRING_LOCKED_ERR);
         }
 
-        validateUser(target);
+        com.github.onsdigital.zebedee.json.Keyring targetCache = getCachedUserKeyring(target);
+
+        for (CollectionDescription desc : assignments) {
+            if (!srcCache.keySet().contains(desc.getId())) {
```

Comment:
Yes. This has happened in develop somehow recently although I've not been able to reproduce it and cannot eliminate it being related to a deployment or something.

When assigning the keys its the keyring of the user creating the new user that is used to populate the new users keyring. If this is missing keys it will spread - Im not sure what we could do to fix this but we would want to know its broken.